pymeasure/instruments/anritsu/anritsuMS464xB.py

```diff
@@ -0,0 +1,912 @@
+
+    service_request_enable_bits = Instrument.control(
+        "*SRE?", "*SRE %d",
+        """Control the Service Request Enable Register bits.
+
+        Valid values are between 0 and 255; setting 0 performs a register reset. Refer to the
+        instrument manual for an explanation of the bits.
+        """,
+        values=[0, 255],
+        validator=strict_range,
+        cast=int,
+    )
+
+    def return_to_local(self):
+        """ Returns the instrument to local operation. """
+        self.write("RTL")
+
+    binary_data_byte_order = Instrument.control(
+        ":FORM:BORD?", ":FORM:BORD %s",
+        """Control the binary numeric I/O data byte order.
+
+        valid values are:
+
+        =====   =========================================
+        value   description
+        =====   =========================================
+        NORM    The most significant byte (MSB) is first
+        SWAP    The least significant byte (LSB) is first
+        =====   =========================================
+        """,
+        values=["NORM", "SWAP"],
+        validator=strict_discrete_set,
+    )
+
+    max_number_of_points = Instrument.control(
+        ":SYST:POIN:MAX?", ":SYST:POIN:MAX %d",
+        """Control the maximum number of points the instrument can measure in a sweep.
+
+        Note that when this value is changed, the instrument will be rebooted.
+        Valid values are 25000 and 100000. When 25000 points is selected, the instrument supports 16
+        channels with 16 traces each; when 100000 is selected, the instrument supports 1 channel
+        with 16 traces.
+        """,
+        values=[25000, 100000],
+        validator=strict_discrete_set,
+        cast=int,
+    )
+
+    number_of_ports = Instrument.measurement(
+        ":SYST:PORT:COUN?",
+        """Get the number of instrument test ports. """,
+        cast=int,
+    )
+
+    number_of_channels = Instrument.control(
+        ":DISP:COUN?", ":DISP:COUN %d",
+        """Control the number of displayed (and therefore accessible) channels.
+
+        When the system is in 25000 points mode, the number of channels can be 1, 2, 3, 4, 6, 8, 9,
+        10, 12, or 16; when the system is in 100000 points mode, the system only supports 1 channel.
+        If a value is provided that is not valid in the present mode, the instrument is set to the
+        next higher channel number.
+        """,
+        values=[1, CHANNELS_MAX],
+        validator=strict_range,
+        cast=int,
+    )
+
+    display_layout = Instrument.control(
+        ":DISP:SPL?", ":DISP:SPL %s",
+        """Control the channel display layout in a Row-by-Column format.
+
+        Valid values are: {}. The number
+        following the R indicates the number of rows, following the C the number of columns.
+        """.format(", ".join(DISPLAY_LAYOUTS)),
+        values=DISPLAY_LAYOUTS,
+        validator=strict_discrete_set,
+        cast=int,
+    )
```

Comment:
Why don't you use a tuple for the values (ok, validation is more difficult)?
```suggestion
    display_layout = Instrument.control(
        ":DISP:SPL?", ":DISP:SPL R%iC%i",
        """Control the channel display layout in a Row-by-Column format.

        Valid values are: {}. The number
        following the R indicates the number of rows, following the C the number of columns.
        """.format(", ".join(DISPLAY_LAYOUTS)),
        cast=int,
    )
```

You cast to int: what does the returned int indicate?

Reply:
Regarding the int: good one, this should just be a string output with the same format as the input.

Regarding the not using a tuple: For most values (e.g. `R2C1`) that would indeed be fine (e.g. `(2, 1)`); but there are a couple of situations where this is a bit more difficult (e.g. `R2C2C1` or `C2R2R1`), where you need a more difficult tuple (e.g. `(2, (2, 1))` and `((2, 1), 2)`. For me it is not worth setting up this conversion/validation for a functionality that I do not expect anyone will very often use (especially not something I expect people to change 'dynamically', but rather always apply a specific certain display layout for a specific measurement type).

Reply:
Oh, I did not know, that you could use more than one R or C (for example your `R2C2C1`).

In that case, I'd leave the code as is (excecpt for a `cast=str`) and maybe one or two examples in the docstring.

Reply:
That is a good idea indeed; I've added the examples (and the cast=str) to both `display_layout` properties.